lib/rubocop/rspec/top_level_group.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module RSpec
+    # Helper methods for top level example group cops
+    module TopLevelGroup
+      extend NodePattern::Macros
+
+      def_node_matcher :example_or_shared_group?,
+                       (
+                         Language::ExampleGroups::ALL +
+                         Language::SharedGroups::ALL
+                       ).block_pattern
+
+      def on_block(node)
+        return unless respond_to?(:on_top_level_group)
+        return unless top_level_group?(node)
+
+        on_top_level_group(node)
+      end
+
+      private
+
+      def top_level_group?(node)
+        top_level_groups.include?(node)
+      end
+
+      def top_level_groups
+        @top_level_groups ||= \
+          top_level_nodes.select { |n| example_or_shared_group?(n) }
+      end
+
+      def top_level_nodes
+        @top_level_nodes ||= \
+          if root_node.begin_type?
+            root_node.children
+          else
+            [root_node]
+          end
+      end
```

Comment:
@pirj 

TBH I am not sure whether this logic is correct. I just noticed there are two cases:
- when there is only one top-level `describe` in the file - the root node (`processed_source.ast`) is a `block` node
- when there are several top-level `describe` blocks - the root node is `block` and all the `describe`s are nested in it as `block` nodes.

Not sure whether there are no other cases.

I check there `block` node type. But I noticed there is a `kwblock` node type and I have no idea when it's used and should we take it into account.

Reply:
Looks like `kwbegin`  means just explicit `begin`/`end` block